### PR TITLE
Graphql docs

### PR DESCRIPTION
**Pull request by kdorosh**
# Description

Add docs and small test fixes for graphql in gloo edge

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/guides/traffic_management/request_processing/graphql/_index.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+And get the following response:
+```json
+{"data":{"pets":[{"name":"Dog"},{"name":"Cat"}]}}
+```
+
+Remember that the REST request returned to our graphql server was the following json:
+```json
+[{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]
+```
+Note that in our response we trimmed this to only the fields we queried for! (the name)
```
Comment (saiskee) [inline docs/content/guides/traffic_management/request_processing/graphql/_index.md:113]:
```suggestion
Here we can see one advantage of using GraphQL over querying the upstream directly. We issued a GraphQL query for only the name of the pets, and though the upstream could return a lot more data, GraphQL was able to filter out only the data which was relevant to our query. In this case, we queried for the names of the pets, and we got back only the names of the pets. 

Read more about the advantages of using GraphQL [here](https://www.apollographql.com/docs/intro/benefits/#graphql-provides-declarative-efficient-data-fetching).
```

**Review by saiskee — APPROVED** [review]
